### PR TITLE
feat: rename functions and refactor native library calls into separate functions

### PR DESCRIPTION
**Pull request by deepjyoti30-st**
## Summary of change

This PR does the following:

- renames the function names for webauthn for them to be more descriptive to the user.
- refactors the native library calls to make them override-able

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `webJsInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: lib/ts/recipe/webauthn/index.ts [hunk 1]
```diff
@@ -311,7 +357,7 @@ export default class RecipeWrapper {
      *
      * @returns `{ status: "OK", ...}` if successful along a description of the user details (id, etc.) and email
      */
-    static registerAndSignUp(input: { email: string; options?: RecipeFunctionOptions; userContext: any }): Promise<
+    static registerUserWithSignUp(input: { email: string; options?: RecipeFunctionOptions; userContext: any }): Promise<
```
Comment (victorbojica) [inline lib/ts/recipe/webauthn/index.ts:360]: I think this name sounds a bit confusing. What would be the difference between register and signUp ?

File: lib/ts/recipe/webauthn/index.ts [hunk 2]
```diff
@@ -358,7 +405,11 @@ export default class RecipeWrapper {
      *
      * @returns `{ status: "OK", ...}` if successful along a description of the user details (id, etc.) and email
      */
-    static authenticateAndSignIn(input: { email: string; options?: RecipeFunctionOptions; userContext: any }): Promise<
+    static authenticateUserWithSignIn(input: {
```
Comment (victorbojica) [inline lib/ts/recipe/webauthn/index.ts:408]: Same here. Sounds a bit confusing

File: lib/ts/recipe/webauthn/recipeImplementation.ts [hunk 3]
```diff
@@ -354,9 +354,29 @@ export default function getRecipeImplementation(
                 fetchResponse,
             };
         },
-        registerAndSignUp: async function ({ email, options, userContext }) {
+        registerUser: async function ({ registrationOptions }) {
```
Comment (victorbojica) [inline lib/ts/recipe/webauthn/recipeImplementation.ts:357]: i think this should contain some credential reference in the name, as the main purpose of this method is to generate the credential  for a user (in order for the credential to be registered - either on sign up or, in the future, on registerCredential in their "profile" )

File: lib/ts/recipe/webauthn/recipeImplementation.ts [hunk 4]
```diff
@@ -371,56 +391,63 @@ export default function getRecipeImplementation(
...
-            let registrationResponse: RegistrationResponseJSON;
-            try {
-                registrationResponse = await startRegistration({ optionsJSON: registrationOptions });
-            } catch (error: any) {
-                if (error.name === "InvalidStateError") {
-                    return { status: "AUTHENTICATOR_ALREADY_REGISTERED" };
-                }
-
-                throw error;
+            const registerUserResponse = await this.registerUser({ registrationOptions });
```
Comment (victorbojica) [inline lib/ts/recipe/webauthn/recipeImplementation.ts:394]: same here. The name should indicate that this is about the credential and not the user.

File: lib/ts/recipe/webauthn/index.ts (hunk 1 above)
Comment (deepjyoti30-st) [reply lib/ts/recipe/webauthn/index.ts]: Agree with the changes, I have made them.

**Review by victorbojica — APPROVED** [review]

**Pull request merged**